modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/QueryTaskExecutorImpl.java

```diff
@@ -54,7 +55,7 @@ public QueryTaskExecutorImpl(String nodeName) {
     public void start() {
         this.stripedThreadPoolExecutor = new StripedThreadPoolExecutor(
                 4,
-                NamedThreadFactory.threadPrefix(nodeName, "sql-execution-pool"),
+                NamedThreadFactory.threadPrefix(nodeName, "sql-execution-pool" + new Random().nextInt()),
```

Comment:
I see NamedThreadFactory already have counter, why do we need also random here ?
```
    @Override
    public Thread newThread(Runnable r) {
        Thread t = new Thread(r);

        t.setDaemon(this.daemon);
        t.setUncaughtExceptionHandler(exHnd);
        t.setName(this.prefix + counter.getAndIncrement());

        return t;
    }
```